tools/rostopic/src/rostopic/__init__.py

```diff
@@ -177,39 +180,93 @@ def print_hz(self):
             min_delta = min(self.times)
 
             self.last_printed_tn = self.msg_tn
-        print("average rate: %.3f\n\tmin: %.3fs max: %.3fs std dev: %.5fs window: %s"%(rate, min_delta, max_delta, std_dev, n+1))
+
+        return rate, min_delta, max_delta, std_dev, n+1
+
+    def print_hz(self):
+        """
+        print the average publishing rate to screen
+        """
+        if not self.times:
+            return
```

Comment:
If `get_hz` would distinguish this case with a different return value this extra check would not be necessary.